app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt

```diff
@@ -42,7 +43,14 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
                               autocompleteResult: InlineAutocompleteEditText.AutocompleteResult? = null)
     }
 
+    interface BrowserNavigationStateProvider {
+        fun isBackEnabled(): Boolean
+        fun isForwardEnabled(): Boolean
+        fun getCurrentUrl(): String?
```

Comment:
nit: if you made this `var url: String?` right in the interface, you wouldn't need an extra getter function in BrowserFragment.

Reply:
interface setters need to be public, so this didn't work.